config/config_mgmt.py

```diff
@@ -213,6 +220,70 @@ def writeConfigDB(self, jDiff):
 
         return
 
+    def add_module(self, yang_module_str, replace_if_exists=False):
+        """
+        Validate and add new YANG module to the system.
+
+        Parameters:
+            yang_module_str (str): YANG module in string representation.
+
+        Returns:
+            None
+        """
+
+        module_name = self.get_module_name(yang_module_str)
+        module_path = os.path.join(YANG_DIR, '{}.yang'.format(module_name))
+        if os.path.exists(module_path) and not replace_if_exists:
+            raise Exception('{} already exists'.format(module_name))
+        with open(module_path, 'w') as module_file:
```

Comment:
Might be a good idea write to temp file and upon success, replace existing file. Else a good file could get removed for a bad i/p.

Reply:
I noticed that I and not using replace_if_exists anyway. So just removed it. Now, we can add module only if it does not exists already.